src/main.rs

```diff
@@ -140,6 +214,26 @@ async fn main() -> ExitCode {
         config
     );
 
+    if let Some(Command::Clean { provision, logs }) = opts.command {
+        if provision {
+            match clean_provisioning_status(&config, &vm_id) {
+                Ok(_) => tracing::info!("Provisioning state cleared."),
+                Err(e) => {
+                    tracing::error!("Failed to clear provisioning state: {e:?}")
+                }
+            }
+        }
+
+        if logs {
+            match clean_log_file(&config) {
```

Comment:
don't log anything here as it should already be done in clean_log_file().

Exit with failure code instead of success on error.